app/src/main/java/net/squanchy/tweets/TweetsPageView.java

```diff
@@ -69,32 +73,40 @@ protected void onAttachedToWindow() {
         initList();
     }
 
+    @Override
+    protected void onDetachedFromWindow() {
+        super.onDetachedFromWindow();
+        subscription.dispose();
+    }
+
     private void initList() {
```

Comment:
You have the (activity) context in the ctor as a parameter... A view cannot be attached to an activity that didn't create it. Also yes please implement the interface, I had _absolutely_ forgotten about its existence 💯